apis/sgv2-restapi/src/test/java/io/stargate/sgv2/it/RestApiV2QRowGetIT.java

```diff
@@ -0,0 +1,948 @@
+
+    final List<Map<String, Object>> allRows = new ArrayList<>();
+    // Get first page
+    final String path = endpointPathForAllRows(testKeyspaceName(), tableName);
+    String response =
+        givenWithAuth()
+            .queryParam("page-size", 2)
+            .when()
+            .get(path)
+            .then()
+            .statusCode(HttpStatus.SC_OK)
+            .extract()
+            .asString();
+    ListOfMapsGetResponseWrapper wrapper = readJsonAs(response, ListOfMapsGetResponseWrapper.class);
+    assertThat(wrapper.getCount()).isEqualTo(2);
+    String pageState = wrapper.getPageState();
+    assertThat(pageState).isNotEmpty();
+    assertThat(wrapper.getData()).hasSize(2);
+    allRows.addAll(wrapper.getData());
+
+    // Then second
+    response =
+        givenWithAuth()
+            .queryParam("page-size", 2)
+            .queryParam("page-state", pageState)
+            .when()
+            .get(path)
+            .then()
+            .statusCode(HttpStatus.SC_OK)
+            .extract()
+            .asString();
+    wrapper = readJsonAs(response, ListOfMapsGetResponseWrapper.class);
+    assertThat(wrapper.getCount()).isEqualTo(2);
+    pageState = wrapper.getPageState();
+    assertThat(pageState).isNotEmpty();
+    assertThat(wrapper.getData()).hasSize(2);
+    allRows.addAll(wrapper.getData());
+
+    // Now no more pages, shouldn't get PagingState either
+    response =
+        givenWithAuth()
+            .queryParam("page-size", 2)
+            .queryParam("page-state", pageState)
+            .when()
+            .get(path)
+            .then()
+            .statusCode(HttpStatus.SC_OK)
+            .extract()
+            .asString();
+    wrapper = readJsonAs(response, ListOfMapsGetResponseWrapper.class);
+    assertThat(wrapper.getPageState()).isNull();
+    assertThat(wrapper.getCount()).isEqualTo(0);
+    assertThat(wrapper.getData()).hasSize(0);
+
+    assertThat(new LinkedHashSet(allRows)).isEqualTo(new LinkedHashSet(expRows));
+  }
+
+  @Test
+  public void getInvalidWhereClause() {
+    final String tableName = testTableName();
+    createSimpleTestTable(testKeyspaceName(), tableName);
+
+    final String rowIdentifier = UUID.randomUUID().toString();
+    insertRow(testKeyspaceName(), tableName, map("id", rowIdentifier));
+
+    String whereClause = "{\"invalid_field\":{\"$eq\":\"test\"}}";
+    final String path = endpointPathForRowGetWith(testKeyspaceName(), tableName);
+    String response =
+        givenWithAuth()
+            .queryParam("where", whereClause)
+            .when()
+            .get(path)
+            .then()
+            .statusCode(HttpStatus.SC_BAD_REQUEST)
+            .extract()
+            .asString();
+    ApiError error = readJsonAs(response, ApiError.class);
+    assertThat(error.code()).isEqualTo(HttpStatus.SC_BAD_REQUEST);
+    assertThat(error.description()).contains("Invalid 'where' parameter, problem: ");
```

Comment:
Maybe this could also check on what the problem is?

Reply:
I was thinking of that, but I think the actual description seemed gRPC-specific and wasn't sure if to include it. But let me go back and double-check, either adding a comment or added check.

Reply:
There is a real error message, will check that. Just need to see if it's persistence-dependant (different on C*3 vs c*4 etc).

Reply:
aaand of course it is slightly different. So need to tune the match a bit.